src/megaclient.cpp

```diff
@@ -549,6 +549,23 @@ Node *MegaClient::getrootnode(Node *node)
     return n;
 }
 
+bool MegaClient::isPrivateNode(handle h)
+{
+    Node *node = nodebyhandle(h);
+    if (!node)
+    {
+        return false;
+    }
+
+    handle rootnode = getrootnode(node)->nodehandle;
+    return (rootnode == rootnodes[0] || rootnode == rootnodes[1] || rootnode == rootnodes[2]);
+}
+
+bool MegaClient::isForeignNode(handle h)
+{
+    return !isPrivateNode(h);
```

Comment:
this one will return true if the node for the handle can't be found, though - shouldn't it return false in that case?  thanks

Reply:
Good catch!! thanks, we will adjust it and merge. It's a minor thing :)